subsys/suit/events/src/suit_events.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <suit_events.h>
+#include <zephyr/kernel.h>
+
+/* Helper mask, with 0th bits for all processors set. */
+#define ALL_CPUS_MASK 0x11111111
+
+/* Helper mask, with all event bits for a given processor set. */
+#define ALL_EVENT_MASK 0x0F
+
+/* The macimum CPU ID handled by this module. */
+#define CPU_N 8
+
+/* The number of bits per CPU inside the event mask. */
+#define BITS_N 4
+
+/* Internal event structure, common for all API calls. */
+static K_EVENT_DEFINE(suit_events);
+
+/* Convert (cpu_id, event mask) pair into global (common for all CPUs) event mask. */
+static uint32_t cpu_events_to_suit_events_mask(uint8_t cpu_id, uint8_t events)
+{
+	/* Check for valid CPU ID values. */
+	if ((cpu_id >= CPU_N) && (cpu_id != SUIT_EVENT_CPU_ID_ANY)) {
+		return 0;
+	}
+
+	/* Mask all unsupported events. */
+	events &= ALL_EVENT_MASK;
+
+	if (cpu_id == SUIT_EVENT_CPU_ID_ANY) {
+		return events * ALL_CPUS_MASK;
+	}
+
+	return (uint32_t)events << (cpu_id * BITS_N);
+}
+
+/* Convert global (common for all CPUs) event mask into an event mask for a given CPU. */
+static uint8_t suit_events_mask_to_cpu_events(uint8_t cpu_id, uint32_t events)
+{
+	uint32_t cpu_shift = 0;
+	uint32_t cpu_events = 0;
+
+	/* Check for valid CPU ID values. */
+	if ((cpu_id >= CPU_N) && (cpu_id != SUIT_EVENT_CPU_ID_ANY)) {
+		return 0;
+	}
+
+	if (cpu_id == SUIT_EVENT_CPU_ID_ANY) {
+		uint8_t common_mask = 0;
+
+		for (size_t id = 0; id < CPU_N; id++) {
+			cpu_shift = (id * BITS_N);
+			cpu_events =
+				(events & ((uint32_t)ALL_EVENT_MASK << cpu_shift)) >> cpu_shift;
+			common_mask |= (cpu_events & ALL_EVENT_MASK);
+		}
+
+		return common_mask;
+	}
+
+	cpu_shift = (cpu_id * BITS_N);
+	cpu_events = (events & ((uint32_t)ALL_EVENT_MASK << cpu_shift)) >> cpu_shift;
+
+	return (cpu_events & ALL_EVENT_MASK);
+}
+
+uint32_t suit_event_post(uint8_t cpu_id, uint8_t events)
```

Comment:
Shouldn't we use `suit_plat_err_t` as a return value? I understand we simply translate the value returned by `k_event_post` but do we plan to use it in any way?